docker/build/xdiff.py

```diff
@@ -120,43 +122,122 @@ def diff_strings(string1, string2, file1, file2):
             elif line.startswith("---"):
                 if not args.quiet and not args.no_meta:
-                    print(term.bold(line), end="")
+                    # print(term.bold(line), end="")
+                    removed_lines.append([line])
             elif line.startswith("+"):
                 if not args.quiet:
-                    print(term.green(line), end="")
+                    # print(term.green(line), end="")
+                    added_lines.append([line])
             elif line.startswith("-"):
                 if not args.quiet:
-                    print(term.red(line), end="")
-            else:
-                if not args.quiet:
-                    print(line, end="")
+                    # print(term.red(line), end="")
+                    removed_lines.append([line])
+            # else:
+            #     if not args.quiet:
+            #         print(line, end="")
     except KeyboardInterrupt:
         sys.exit(130)
-    if n == 0:
-        exit = 0
-    else:
-        exit = 1
-    return exit
+    # if n == 0:
+    #     exit = 0
+    # else:
+    #     exit = 1
+    
+    parsed_diff = [added_lines, removed_lines]
+    return parsed_diff
+
+def extract_urls_from_diff(parsed_diff):
+  extracted_urls = []
+  for line in parsed_diff:
+      for str in line:
+          search = re.search("<loc>(.+?)</loc>", str)
+          if (search):
+              url = search.group(1)
+              parsed_url = url.replace('testing.objectiv.io', 'objectiv.io')
+              parsed_url = parsed_url.replace('staging.objectiv.io', 'objectiv.io')
+              extracted_urls.append([parsed_url])
+      
+  return extracted_urls
+
+def write_urls_to_csv(urls, file='./tmp/urls.csv'):
+    header = ['url']
+    file_exists=os.path.exists(file)
+    with open(file, "a" if file_exists else "w", encoding='UTF8', newline='') as f:
+        writer = csv.writer(f)
+        if(not file_exists):
+            # write the header first
+            writer.writerow(header)
+
+        # write multiple rows
+        writer.writerows(urls)
+
+    print("Written to CSV:", file)
+
+def replace_text_in_file(file_to_replace, old_text, new_text):
+    # Read in the file
+    with open(file_to_replace, 'r+') as file:
+        filedata = file.read()
+        # Replace the target string
+        filedata = filedata.replace(old_text, new_text)
+        # Setting the position to the top
+        # of the page to insert data
+        file.seek(0)
+        # Write the file out again
+        file.write(filedata)
+        # Truncating the file size
+        file.truncate()
+
+def main():
+    # replace certain environment lines in the file and write it back
+    replace_text_in_file(args.file1, "testing.objectiv.io", "objectiv.io")
+    replace_text_in_file(args.file2, "testing.objectiv.io", "objectiv.io")
+
+    parsed_diff = parse_diff(args)
+
+    # print and write added URLs
+    if(len(parsed_diff[0]) > 0):
```

Comment:
empty List evaluates to `False`
```suggestion
    if parsed_diff[0]:
```